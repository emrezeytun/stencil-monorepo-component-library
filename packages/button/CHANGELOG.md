# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 1.0.5 (2025-08-24)


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))





## 1.0.4 (2025-08-24)


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))





## 1.0.3 (2025-08-24)


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))





## 1.0.2 (2025-08-24)


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))
