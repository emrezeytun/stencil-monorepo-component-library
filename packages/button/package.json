{
  "name": "@arcadia-components/button",
  "version": "1.0.6",
  "description": "Button component built with Stencil",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.cjs.js",
  "module": "dist/index.js",
  "types": "dist/types/index.d.ts",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "unpkg": "dist/button/button.esm.js",
  "exports": {
    ".": {
      "import": "./dist/button/button.esm.js",
      "require": "./dist/button/button.cjs.js",
      "types": "./dist/types/index.d.ts"
    },
    "./button": {
      "import": "./dist/components/button.js",
      "types": "./dist/components/button.d.ts"
    },
    "./loader": {
      "import": "./loader/index.js",
      "require": "./loader/index.cjs",
      "types": "./loader/index.d.ts"
    },
    "./react": {
      "import": "./bindings/react/index.js",
      "types": "./bindings/react/index.d.ts"
    },
    "./vue": {
      "import": "./bindings/vue/index.js",
      "types": "./bindings/vue/index.d.ts"
    },
    "./bindings/vue": {
      "import": "./bindings/vue/index.js",
      "types": "./bindings/vue/index.d.ts"
    },
    "./bindings/vue/index.js": {
      "import": "./bindings/vue/index.js",
      "types": "./bindings/vue/index.d.ts"
    },
    "./angular": {
      "import": "./bindings/angular/index.js",
      "types": "./bindings/angular/index.d.ts"
    }
  },
  "files": [
    "dist/",
    "loader/",
    "bindings/"
  ],
  "scripts": {
    "build": "stencil build",
    "start": "stencil build --dev --watch --serve",
    "test": "stencil test --spec",
    "test.watch": "stencil test --spec --watchAll",
    "generate": "stencil generate",
    "storybook": "storybook dev -p 6009",
    "build-storybook": "storybook build"
  },
  "keywords": [
    "stencil",
    "web-components",
    "button",
    "ui",
    "component",
    "react",
    "vue",
    "angular",
    "arcadia-components"
  ],
  "author": "Arcadia Components",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/your-org/arcadia-components.git",
    "directory": "packages/button"
  },
  "homepage": "https://github.com/your-org/arcadia-components#readme",
  "bugs": {
    "url": "https://github.com/your-org/arcadia-components/issues"
  },
  "devDependencies": {
    "@stencil/core": "^4.27.1",
    "@types/jest": "^29.5.14",
    "@types/node": "^22.13.5",
    "jest": "^29.7.0",
    "jest-cli": "^29.7.0",
    "puppeteer": "^24.3.0",
    "typescript": "^5.3.0"
  },
  "gitHead": "8cb0519888418c79da209e1b2807e05e1dc31cc6"
}
