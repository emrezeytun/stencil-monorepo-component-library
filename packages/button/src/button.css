.button {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 6px;
  font-family: inherit;
  font-weight: 500;
  cursor: pointer;
  transition: all 0.2s ease;
  text-decoration: none;
  box-sizing: border-box;
}

/* Sizes */
.button--small {
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.4;
}

.button--medium {
  padding: 12px 20px;
  font-size: 16px;
  line-height: 1.5;
}

.button--large {
  padding: 16px 24px;
  font-size: 18px;
  line-height: 1.6;
}

/* Variants */
.button--primary {
  background-color: #007bff;
  color: white;
}

.button--primary:hover:not(.button--disabled) {
  background-color: #0056b3;
  transform: translateY(-1px);
}

.button--secondary {
  background-color: #6c757d;
  color: white;
}

.button--secondary:hover:not(.button--disabled) {
  background-color: #545b62;
  transform: translateY(-1px);
}

.button--danger {
  background-color: #dc3545;
  color: white;
}

.button--danger:hover:not(.button--disabled) {
  background-color: #c82333;
  transform: translateY(-1px);
}

/* Disabled state */
.button--disabled {
  opacity: 0.6;
  cursor: not-allowed;
  transform: none !important;
}

.button:focus {
  outline: 2px solid #007bff;
  outline-offset: 2px;
}
