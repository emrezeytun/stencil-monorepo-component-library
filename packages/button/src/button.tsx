import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';

@Component({
  tag: 'comp-button',
  styleUrl: 'button.css',
  shadow: true,
})
export class Button {
  @Prop() variant: 'primary' | 'secondary' | 'danger' = 'primary';
  @Prop() size: 'small' | 'medium' | 'large' = 'medium';
  @Prop() disabled: boolean = false;
  @Prop() type: 'button' | 'submit' | 'reset' = 'button';

  @Event() buttonClick: EventEmitter<MouseEvent>;

  private handleClick = (event: MouseEvent) => {
    if (!this.disabled) {
      this.buttonClick.emit(event);
    }
  };

  render() {
    return (
      <button
        class={`button button--${this.variant} button--${this.size} ${this.disabled ? 'button--disabled' : ''}`}
        disabled={this.disabled}
        type={this.type}
        onClick={this.handleClick}
      >
        test 123
        <slot></slot>
      </button>
    );
  }
}
