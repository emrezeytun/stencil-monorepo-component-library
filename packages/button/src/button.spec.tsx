import { newSpecPage } from '@stencil/core/testing';
import { Button } from './button';

describe('comp-button', () => {
  it('renders without crashing (smoke test)', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button>Test</comp-button>`,
    });
    
    expect(page.root).toBeTruthy();
    expect(page.root?.tagName).toBe('COMP-BUTTON');
  });

  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button>Click me</comp-button>`,
    });
    
    const button = page.root?.shadowRoot?.querySelector('button');
    expect(button).toBeTruthy();
    // Note: Slot content might not be available in unit tests
    expect(button?.type).toBe('button');
    expect(button?.hasAttribute('disabled')).toBe(false);
  });

  it('applies variant classes correctly', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button variant="secondary">Test</comp-button>`,
    });
    
    const button = page.root?.shadowRoot?.querySelector('button');
    expect(button?.className).toContain('button--secondary');
    expect(button?.className).toContain('button--medium'); // default size
  });

  it('applies size classes correctly', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button size="large">Test</comp-button>`,
    });
    
    const button = page.root?.shadowRoot?.querySelector('button');
    expect(button?.className).toContain('button--large');
    expect(button?.className).toContain('button--primary'); // default variant
  });

  it('handles disabled state', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button disabled>Test</comp-button>`,
    });
    
    const button = page.root?.shadowRoot?.querySelector('button');
    expect(button?.hasAttribute('disabled')).toBe(true);
    expect(button?.className).toContain('button--disabled');
  });

  it('sets correct button type', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<comp-button type="submit">Submit</comp-button>`,
    });
    
    const button = page.root?.shadowRoot?.querySelector('button');
    expect(button?.type).toBe('submit');
  });

  it('matches snapshot for all variants', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
        <comp-button variant="primary">Primary</comp-button>
        <comp-button variant="secondary">Secondary</comp-button>
        <comp-button variant="danger">Danger</comp-button>
      `,
    });
    
    expect(page.root).toMatchSnapshot();
  });
});
