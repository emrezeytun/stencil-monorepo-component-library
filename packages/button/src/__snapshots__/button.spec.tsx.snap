// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp-button matches snapshot for all variants 1`] = `
<comp-button variant="primary">
  <template shadowrootmode="open">
    <button class="button button--medium button--primary" type="button">
      <slot></slot>
    </button>
  </template>
  Primary
</comp-button>
`;
